bpdm-gate/src/test/kotlin/org/eclipse/tractusx/bpdm/gate/controller/BusinessPartnerControllerIT.kt

```diff
@@ -0,0 +1,225 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdm.gate.controller
+
+import org.assertj.core.api.Assertions
+import org.eclipse.tractusx.bpdm.common.dto.ClassificationDto
+import org.eclipse.tractusx.bpdm.common.dto.request.PaginationRequest
+import org.eclipse.tractusx.bpdm.gate.api.client.GateClient
+import org.eclipse.tractusx.bpdm.gate.api.model.BusinessPartnerIdentifierDto
+import org.eclipse.tractusx.bpdm.gate.api.model.BusinessPartnerStateDto
+import org.eclipse.tractusx.bpdm.gate.api.model.request.BusinessPartnerInputRequest
+import org.eclipse.tractusx.bpdm.gate.api.model.response.BusinessPartnerInputDto
+import org.eclipse.tractusx.bpdm.gate.util.CommonValues
+import org.eclipse.tractusx.bpdm.gate.util.DbTestHelpers
+import org.eclipse.tractusx.bpdm.gate.util.PostgreSQLContextInitializer
+import org.eclipse.tractusx.bpdm.gate.util.RequestValues
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.HttpStatus
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import java.time.Instant
+
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    properties = ["bpdm.api.upsert-limit=3"]
+)
+@ActiveProfiles("test")
+@ContextConfiguration(initializers = [PostgreSQLContextInitializer::class])
+class BusinessPartnerControllerIT @Autowired constructor(
+    val testHelpers: DbTestHelpers,
+    val gateClient: GateClient,
+) {
+
+    @BeforeEach
+    fun beforeEach() {
+        testHelpers.truncateDbTables()
+    }
+
+    @Test
+    fun `insert minimal business partner`() {
+        val upsertRequests = listOf(RequestValues.bpInputRequestMinimal)
+        val upsertResponses = gateClient.businessParters().upsertBusinessPartnersInput(upsertRequests).body!!
+        assertUpsertResponsesMatchRequests(upsertResponses, upsertRequests)
+
+        val searchResponsePage = gateClient.businessParters().getBusinessPartnersInput(null)
+        assertEquals(1, searchResponsePage.totalElements)
+        testHelpers.assertRecursively(searchResponsePage.content).isEqualTo(upsertResponses)
+    }
+
+    @Test
+    fun `insert two business partners`() {
+        val upsertRequests = listOf(RequestValues.bpInputRequestDefault, RequestValues.bpInputRequestMinimal)
```

Comment:
I like the naming of the data 'bpInputRequestMinimal'
But hte name bpInputRequestDefault is not so meaningful like bpInputRequestMinimal.
Does it make sense to name it bpInputRequestMaximal.

Perhaps we can add a testCase for non-latin data like chinese.

Reply:
Sounds good